src/query/storages/fuse/src/operations/commit.rs

```diff
@@ -161,16 +161,11 @@ impl FuseTable {
         .await;
         if need_to_save_statistics {
             let table_statistics_location: String = table_statistics_location.unwrap();
-            match &res {
-                Ok(_) => TableSnapshotStatistics::cache().put(
+            if res.is_ok() {
+                TableSnapshotStatistics::cache().put(
                     table_statistics_location,
                     Arc::new(table_statistics.unwrap()),
-                ),
-                Err(e) => {
-                    if Self::no_side_effects_in_meta_store(e) {
-                        let _ = operator.delete(&table_statistics_location).await;
-                    }
-                }
+                )
```

Comment:
Do we need a error log for `else`?

Reply:
let's log the error. there might be some duplication (of error logs by caller site), but not too much